Dynatrace/dynatrace-operator
# Refactors the troubleshoot command to cleanup code

# Description

Refactors the troubleshoot command to:
1. Remove duplicated code
2. Refactor code to make it more readable
3. Remove code that is unused due to refactorings

Also introduces some minor improvements that where made as part of refactoring the code:
* Adds a `warning` log level for log messages signifying a potential error
* Changes the character used for success messages from a square root to a checkmark
* Changes the character used for errors to a bigger `x`
* Adds green, red, and yellow color to success, error, and warning messages respectively.

## How can this be tested?

* Create different Dynakube resources. E.g.:
  * a simple dynakube
  * one with a custom pullsecret
  * one with a proxy
  * using a codemodules image
  * covering different error cases
* Deploy the operator and run the troubleshoot command
* Check that the correct state of the dynakube is determined by the command


## Checklist
